File: samples/client/petstore/java/feign/src/main/java/org/openapitools/client/api/DefaultApi.java
package org.openapitools.client.api;

import org.openapitools.client.ApiClient;
import org.openapitools.client.EncodingUtils;
import org.openapitools.client.model.ApiResponse;

import org.openapitools.client.model.InlineResponseDefault;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import feign.*;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
public interface DefaultApi extends ApiClient.Api {


  /**
   * 
   * 
   * @return InlineResponseDefault
   */
  @RequestLine("GET /foo")
  @Headers({
    "Accept: application/json",
  })
  InlineResponseDefault fooGet();

  /**
   * 
   * Similar to <code>fooGet</code> but it also returns the http response headers .
   * 
   * @return A ApiResponse that wraps the response boyd and the http headers.
   */
  @RequestLine("GET /foo")
  @Headers({
    "Accept: application/json",
  })
  ApiResponse<InlineResponseDefault> fooGetWithHttpInfo();


}

File: samples/openapi3/client/petstore/kotlin-jvm-retrofit2-coroutines/src/main/kotlin/org/openapitools/client/apis/DefaultApi.kt
package org.openapitools.client.apis

import org.openapitools.client.infrastructure.CollectionFormats.*
import retrofit2.http.*
import retrofit2.Response
import okhttp3.RequestBody

import org.openapitools.client.models.InlineResponseDefault

interface DefaultApi {
    /**
     * 
     * 
     * Responses:
     *  - 0: response
     * 
     * @return [InlineResponseDefault]
     */
    @GET("foo")
    suspend fun fooGet(): Response<InlineResponseDefault>

}
